### ISSUE TITLE
Slowness with default flannel setup - slirp4netns or mtu?

### ISSUE THREAD

**Issue by vsoch** [rootless-containers/usernetes#319]
Hey @AkihiroSuda ! I was doing some tests on my tiny cluster and noticed quite a bit of slowness with usernetes vs. without - here is a simple run of iperf3 for two metrics:

![client_to_server_bitrate_gbits_persecond](https://github.com/rootless-containers/usernetes/assets/814322/1bb76019-cd3a-4a3a-88c9-ee8a5fb5b5cb)
![client_to_server_transfer_gbytes](https://github.com/rootless-containers/usernetes/assets/814322/0258852e-fc84-4372-9583-4ecbbeb40561)

I had this assumption in my head that slirp4netns was to blame - I read something somewhere about it (that I can't find now) that it would be a bottleneck (is that true)? But then I was looking here: https://github.com/flannel-io/flannel/blob/e8fb8108622bb9646dc7de84df19adbae319acb8/pkg/subnet/subnet.go#L104 and it seems the default mtu value is 1450 - could that explain the slowness (or something else)? Is there a way to make the usernetes network performance equal to the bare metal? Thank you!

**Comment by vsoch** [comment]
@AkihiroSuda so you would say the slowness isn't isn't mtu, but still something related to slirp4netns (could you tell me in layman's terms the issue)? I am putting together a talk and want to mention why the network was slow, at least at a high level. If it's just the mtu value I could also try the experiments again. I'm trying to understand why it's slow.

**Comment by AkihiroSuda** [comment]
It is still worth trying to adjust MTU, but won't reach 10 Gbps.

**Comment by AkihiroSuda** [comment]
You may also refer to https://pibvt.net/IPSJ-OS22156009.pdf to grab the design of slirp4netns, and how its overhead can be eliminated with bypass4netns. This revision does not support VXLAN though.

**Comment by vsoch** [comment]
okay confirmed the default is indeed 1450:

```console
# cat /run/flannel/subnet.env 
FLANNEL_NETWORK=10.244.0.0/16
FLANNEL_SUBNET=10.244.0.1/24
FLANNEL_MTU=1450
FLANNEL_IPMASQ=true
```
Trying to understand how to change it next.

**Comment by vsoch** [comment]
@AkihiroSuda could it be mtu related if we are also using the network with mtu 1500 for the comparison case? For example, see the middle one here:

![image](https://github.com/rootless-containers/usernetes/assets/814322/8775de88-0dd3-4da7-93b3-f4dc6709b740)

And then that flux is using it for the overlay network (see the name under default bind) 
![image](https://github.com/rootless-containers/usernetes/assets/814322/dd12f3b1-7e9a-4f0c-a0a6-b21e43448027)

If they both are 1500 (but flux is still much faster) that probably can't be the causative factor.

**Comment by AkihiroSuda** [comment]
You may rather try to decrease this MTU https://github.com/moby/moby/blob/afc7e581e601d53d3b346828bd94ac1fed1e226d/contrib/dockerd-rootless.sh#L13

**Comment by vsoch** [comment]
Should be good here thanks for the help!
